### DISCUSSION TITLE
Support for TV specific file extensions (ios.tv.js, android.tv.js)

### DISCUSSION THREAD

**Discussion by henninghall** [react-native-tvos/react-native-tvos#630] [Feature Requests]
Is it possible to use specific file extensions for tv components, for instance `file.tv.js`, similar to what android and iOS uses with `.ios.js` and `.android.js`? I couldn't find any discussions about this. Thanks!

**Comment by douglowder** [comment]
That hasn't been implemented yet. Interesting idea. We would also need to support `*.tv.ios.js`, etc.

**Comment by Barteque** [comment]
Hi @douglowder 
What is the status of that? 
Can we use tv.ios. or .tvos. files?

**Comment by henninghall** [comment]
What we ended up doing was a workaround similar to this

`player.tsx`
`playerTvOS.tsx`
`playerMobile.tsx`

<hr/>

**player.tsx**

```
const Player = Platform.OS === "tvOS"
  ? (require('./PlayerTvOS') as typeof import('./PlayerTvOS')).default
  : (require('./PlayerMobile') as typeof import('./PlayerMobile')).default;
```

**Comment by douglowder** [comment]
I've moved this to the "Feature Requests" section of discussions. Thanks to @okwasniewski, @thymikee and others working on Vision OS, for their recent work on custom Metro resolvers for out of tree platforms. I've now got a resolver working that will implement this feature. I'll post progress reports here.

**Comment by douglowder** [comment]
After looking at a number of ways to address this, it is safest to provide an example for developers to follow, and documentation on how it works. See #640 .

**Comment by douglowder** [comment]
Completed and released in 0.73.2-0.
